### PR TITLE
support rename for login, user and a few other types

### PR DESCRIPTION
**Pull request by alanrenmsft**
Support renaming for login and user, also enabled for Database, View, table column object types.

STS changes:
image

Rename UX:
rename-object

during testing I noticed the following issues:
1.  in object explorer (connections view) for table column, the parent node won't refresh after renaming  (https://github.com/microsoft/azuredatastudio/issues/22343)
2.  in azure view, the refresh doesn't work for any type. (https://github.com/microsoft/azuredatastudio/issues/22343)

### CODE REVIEW

File: extensions/mssql/package.json [hunk 1]
```diff
@@ -551,6 +570,16 @@
           "when": "connectionProvider == MSSQL && nodeType =~ /^(ServerLevelLogin|User)$/ && config.workbench.enablePreviewFeatures",
           "group": "connection@2"
         },
+        {
+          "command": "mssql.renameObject",
+          "when": "connectionProvider == MSSQL && nodeType =~ /^(Database|ServerLevelLogin|User|Table|View)$/ && config.workbench.enablePreviewFeatures",
+          "group": "connection@3"
+        },
+        {
+          "command": "mssql.renameObject",
```
Comment (alanrenmsft) [inline extensions/mssql/package.json:579]: Column is also available in view which doesn't support renaming, so only enable it for table columns.

File: extensions/mssql/src/objectManagement/commands.ts [hunk 2]
```diff
@@ -165,6 +168,58 @@ async function handleDeleteObjectCommand(context: azdata.ObjectExplorerContext,
...
+				return localizedConstants.NameCannotBeEmptyError;
+			} else {
+				// valid
+				return undefined;
+			}
+		}
+	});
+
+	// return if no change was made or the dialog was canceled.
+	if (newName === context.nodeInfo.label || !newName) {
```
Comment (caohai) [inline extensions/mssql/src/objectManagement/commands.ts:192]: Is `context.nodeInfo.label` the same as `originalName`?

Comment (alanrenmsft) [reply extensions/mssql/src/objectManagement/commands.ts:192]: good catch. I'll fix it.

**Review by caohai — APPROVED** [review]

File: extensions/mssql/src/objectManagement/ui/objectManagementDialogBase.ts [hunk 3]
```diff
@@ -234,9 +234,9 @@ export abstract class ObjectManagementDialogBase<ObjectInfoType extends ObjectMa
 	protected createGroup(header: string, items: azdata.Component[], collapsible: boolean = true, collapsed: boolean = false): azdata.GroupContainer {
 		return this.modelView.modelBuilder.groupContainer().withLayout({
 			header: header,
-			collapsed: false,
-			collapsible: collapsible
-		}).withProps({ collapsed: collapsed }).withItems(items).component();
+			collapsible: collapsible,
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/objectManagement/ui/objectManagementDialogBase.ts:237]: Is this related to these changes?

**Review by Charles-Gagnon — APPROVED** [review]
@corivera @erinstellato-ms @ssreerama FYI - looks like some of our work is being done for us already 😄

File: extensions/mssql/src/objectManagement/commands.ts [hunk 4]
```diff
@@ -165,6 +168,58 @@ async function handleDeleteObjectCommand(context: azdata.ObjectExplorerContext,
...
+				});
+			}
+			catch (err) {
+				operation.updateStatus(azdata.TaskStatus.Failed, localizedConstants.RenameObjectError(nodeTypeDisplayName, originalName, newName, getErrorMessage(err)));
+				TelemetryReporter.createErrorEvent2(TelemetryViews.ObjectManagement, TelemetryActions.RenameObject, err).withAdditionalProperties({
+					objectType: context.nodeInfo.nodeType
+				}).send();
+				return;
+			}
+			await refreshParentNode(context);
```
[the remark this answers is not in the document]
Comment (alanrenmsft) [reply extensions/mssql/src/objectManagement/commands.ts:217]: done

File: extensions/mssql/src/objectManagement/ui/objectManagementDialogBase.ts (hunk 3 above)
Comment (alanrenmsft) [reply extensions/mssql/src/objectManagement/ui/objectManagementDialogBase.ts:237]: slide in a small fix, previously the collapsed property is not set properly.
